### ISSUE TITLE
CLI: --require options doesn't work for file with illegal chars for JS variable

### ISSUE THREAD

**Issue by ryoqun** [jashkenas/coffeescript#4209]
The following snippet tells everything...

``` bash
$ echo 'console.log "hi"' > /tmp/hello-world.coffee
$ echo 'console.log "hi"' > /tmp/hello_world.coffee
$ coffee -r /tmp/hello_world.coffee -e 'console.log("eval")'
hi
eval
$ coffee -r /tmp/hello-world.coffee -e 'console.log("eval")'
ReferenceError: hello is not defined
```

**Comment by ryoqun** [comment]
By the way, this is causing the bug in the wild: https://github.com/benbria/coffee-coverage/blob/master/register-istanbul.js

**Comment by ryoqun** [comment]
Hooray for new release!!

So, I'm bumping this just to remind others.. :)

**Comment by GeoffreyBooth** [comment]
Fixed by #4658.
